src/Symfony/Component/Messenger/Transport/InMemoryTransport.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Contracts\Service\ResetInterface;
+
+/**
+ * Transport that stay in memory. Useful for testing purpose.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class InMemoryTransport implements TransportInterface, ResetInterface
+{
+    /**
+     * @var Envelope[]
+     */
+    private $sent = [];
+
+    /**
+     * Is the transport stopped?
+     *
+     * @var bool
+     */
+    private $stopped = false;
+
+    /**
+     * @var Envelope[]
+     */
+    private $acknowledged = [];
+
+    /**
+     * @var Envelope[]
+     */
+    private $rejected = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        if ($this->stopped) {
+            return;
+        }
+
+        foreach ($this->sent as $envelope) {
+            $handler($envelope);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stop(): void
+    {
+        $this->stopped = true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope): Envelope
+    {
+        $this->sent[] = $envelope;
+
+        return $envelope;
+    }
+
+    public function isStopped(): bool
+    {
+        return $this->stopped;
+    }
+
+    public function reset()
+    {
+        $this->sent = $this->rejected = $this->acknowledged = [];
+    }
+
+    public function ack(Envelope $envelope): void
+    {
+        $this->acknowledged[] = $envelope;
+    }
+
+    public function reject(Envelope $envelope): void
+    {
+        $this->rejected[] = $envelope;
+    }
+
+    /**
+     * @return Envelope[]
+     */
+    public function getAcknowledged(): array
+    {
+        return $this->acknowledged;
+    }
+
+    /**
+     * @return Envelope[]
+     */
+    public function getRejected(): array
+    {
+        return $this->rejected;
+    }
+
+    /**
+     * {@inheritdoc}
```

Comment:
Can you move it up, next to the "inherited" methods?

Reply:
Fixed, with all above.